### PR TITLE
Add IP masking function

### PR DESCRIPTION
**Pull request by taktv6**

### CODE REVIEW

File: net/ip.go [hunk 1]
```diff
@@ -309,3 +311,32 @@ func (ip *IP) Next() *IP {
...
+	if ip.isLegacy {
+		ip.maskLastNBitsIPv4(n)
+		return ip
+	}
+
+	ip.maskLastNBitsIPv6(n)
+	return ip
+}
+
+func (ip *IP) maskLastNBitsIPv4(n uint8) {
```
Comment (BarbarossaTM) [inline net/ip.go:328]: I'd add a check if n <= 32 here

File: net/ip.go [hunk 2]
```diff
@@ -309,3 +311,32 @@ func (ip *IP) Next() *IP {
...
+	mask := uint64((math.MaxUint64 << (n)))
+	ip.lower = ip.lower & mask
+}
+
+func (ip *IP) maskLastNBitsIPv6(n uint8) {
```
Comment (BarbarossaTM) [inline net/ip.go:333]: Same as above with <= 128 obviously

File: net/ip.go (hunk 1 above)
Comment (taktv6) [reply net/ip.go:328]: For what? Cutting off more than the address length returns 0.0.0.0 and is what I would expect. No?

**Review by BarbarossaTM — APPROVED** [review]
Yo.

**Pull request merged**